### PR TITLE
Add parity check UI

### PR DESCRIPTION
**Pull request by ethax-ross**
### Context

We want to be able to kick off a parity check from the UI (similar to migrations) and view the progress/results.

### Changes proposed in this pull request

- Add basic UI for parity checks

Add a basic UI to kick off a parity check and view the progress/completion status.

Add `ParityCheckJob` to run parity checks in the background.

Add methods to `ParityCheck` service for tracking progress.

Add `ParityChecksController` and associated views; no details of the check results are surfaced yet, just the functionality to start and view the overall status.

DRY up parity check tests with `migration_helper`.

- Add completion and details to parity check UI

Fleshes out the bare-bones parity check UI with an overview list of the paths called for each provider on the completion page. If the request responses/status codes are different you can click on the path to view a detailed report of the comparison along with a diff of the response bodies. Also includes a basic response time comparison chart that plots the averages and min/max for each path.

Add detailed response comparison action/view.

Add `diffy` to compare response bodies.

Update CSRP to allow chart.js from a CDN (as its only used on this page and it will be deleted shortly I figured this was the best way to include it/ensure its removed).

Remove `equal?` from `ResponseComparison` as this is a system method it started giving issues; we can just negate `different?` in its place.

### Guidance for review

This has been tested on migration; the screenshots below are fabricated to show the diffing though. In migration it flags expected differences on statements (date and the missing `type` attribute from NPQ).

| Start | Running | Complete | Details |
| -- | -- | -- | -- |
| screencapture-0-0-0-0-3000-npq-separation-migration-parity-checks-2024-10-18-08_23_27  |  screencapture-0-0-0-0-3000-npq-separation-migration-parity-checks-2024-10-18-08_24_30 |  screencapture-0-0-0-0-3000-npq-separation-migration-parity-checks-2024-10-18-08_33_10 | screencapture-0-0-0-0-3000-npq-separation-migration-parity-checks-response-comparisons-2-2024-10-18-08_16_14 |

### CODE REVIEW

File: spec/controllers/npq_separation/migration/parity_checks_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+require "rails_helper"
+
+RSpec.describe NpqSeparation::Migration::ParityChecksController, type: :request do
```
Comment (cwrw) [inline spec/controllers/npq_separation/migration/parity_checks_controller_spec.rb:3]: should this be under spec/requests instead?

**Review by cwrw — APPROVED** [review]
it's beautiful :D just a very minor comment

File: spec/controllers/npq_separation/migration/parity_checks_controller_spec.rb (hunk 1 above)
Comment (ethax-ross) [reply spec/controllers/npq_separation/migration/parity_checks_controller_spec.rb:3]: I put it here to be consistent with the migration request spec (which I think was introduced before we started using `spec/requests`), but we did agree to put them there going forward so I'll update 👍 I'll consolidate all the `npq_separation` specs from `/controllers` with `/requests` while I'm at it as there's a few

**Review by cwrw — APPROVED** [review]
thank you!

**Review by leandroalemao — COMMENTED** [review]
Hi @ethax-ross this looks awesome!!! Tks! 👍 🔝

**Review by leandroalemao — APPROVED** [review]

**Review by mooktakim — APPROVED** [review]
Looks great

**Pull request merged**